sdk/servicebus/Azure.Messaging.ServiceBus/src/Transports/Amqp/ServiceBusMessageExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Messaging.ServiceBus.Transports.Amqp
+{
+    /// <summary>
+    /// Extension methods for <see cref="ServiceBusMessage"/>.
+    /// </summary>
+    public static class ServiceBusMessageExtensions
+    {
+        /// <summary>
+        /// Checks if the message contains an AMQP body.
+        /// </summary>
+        /// <param name="message">The message to use.</param>
+        /// <returns>True if the message contains an AMQP body. Otherwise, false.</returns>
+        public static bool HasAmqpBody(this ServiceBusMessage message) => message.TransportBody is AmqpTransportBody;
```

Comment:
When would this be false?

Reply:
Originally, I was using this in internal code. Then I thought, "This might be useful for a user to know." But, yes, it is future proofing at this point. I'll remove it.

Reply:
This has been removed.